Asset/ProvisionedPathResolver.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the EzCoreExtraBundle package.
+ *
+ * (c) Jérôme Vieilledent <jerome@vieilledent.fr>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Lolautruche\EzCoreExtraBundle\Asset;
+
+use Symfony\Component\Finder\Finder;
+
+class ProvisionedPathResolver implements AssetPathResolverInterface, AssetPathProvisionerInterface
+{
+    /**
+     * @var array
+     */
+    private $resolvedPaths;
+
+    /**
+     * @var AssetPathResolverInterface
+     */
+    private $innerResolver;
+
+    /**
+     * @var string
+     */
+    private $webRootDir;
+
+    public function __construct(array $resolvedPaths, AssetPathResolverInterface $innerResolver, $webRootDir)
+    {
+        $this->resolvedPaths = $resolvedPaths;
+        $this->innerResolver = $innerResolver;
+        $this->webRootDir = $webRootDir;
+    }
+
+    /**
+     * Looks for $path within pre-resolved paths for provided design.
+     * If it cannot be found, fallbacks to original resolver.
+     *
+     * {@inheritdoc}
+     */
+    public function resolveAssetPath($path, $design)
+    {
+        if (!isset($this->resolvedPaths[$design][$path])) {
+            return $this->innerResolver->resolveAssetPath($path, $design);
+        }
+
+        return $this->resolvedPaths[$design][$path];
+    }
+
+    public function provisionResolvedPaths(array $assetsPaths, $design)
+    {
+        $webrootDir = $this->webRootDir;
+        $assetsLogicalPaths = [];
+        foreach ($assetsPaths as $path) {
+            $themePath = "$webrootDir/$path";
+            // Exclude "themes/" directory under a theme one, to avoid recursion.
+            // Applies mainly for override directories
+            // e.g. "assets/", which is both an override dir and where app level themes can be defined.
+            /** @var \SplFileInfo $fileInfo */
+            foreach ((new Finder())->files()->in($themePath)->exclude('themes')->followLinks()->ignoreUnreadableDirs() as $fileInfo) {
```

Comment:
You could maybe move this to a private method with a proper name, and move the inline comments as phpdoc of that method ?